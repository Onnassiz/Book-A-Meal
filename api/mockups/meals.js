let meals = [
  {
    id: 1,
    name: 'Chicken Pa-Naeng - Beef',
    description: 'Red Curry with Peanut Sauce, Bamboo Shoot, Ginger, Bell Pepper, Green Bean, Galanga and Lemon Leaves. Choice of Chicken, Beef, Pork, Tofu or Shrimp.',
    price: 2300,
    category: 'Hot Meal',
    image: '1.jpg',
  },
  {
    id: 2,
    name: 'Squash Curry - Tofu',
    description: 'Red and Yellow Curry Coconut Milk with Squash, Potatoes, Green Bean & Bamboo Shoot. Choice of Chicken, Beef, Pork, Tofu or Shrimp.',
    price: 1200,
    category: 'Lunch',
    image: '2.jpg',
  },
  {
    id: 3,
    name: 'Vegetable Fried Rice (Khao Phat Phak)',
    description: 'Stir Fried with Eggs, Chinese Broccoli, Tofu and Onion. Served with Cucumber and Lime.',
    price: 1500,
    category: 'Snack',
    image: '3.jpg',
  },
  {
    id: 4,
    name: 'Shrimp Frid Rice',
    description: 'Stir Fried Rice with Shrimp, Egg, Peas, Corn, Bean, Carrots, Onion. Topped with Cucumber and Lime.',
    price: 2000,
    category: 'Snack',
    image: '4.jpg',
  },
  {
    id: 5,
    name: 'Special Combination Fried Rice',
    description: 'Stir Fried Rice with Beef, Chicken, Pork, Shrimp, Chinese Sausage & Egg',
    price: 1400,
    category: 'Hot Meal',
    image: '5.jpg',
  },
  {
    id: 6,
    name: 'Yum - Pork',
    description: 'Slice Thin Beef Mixed With Lettuce, Thread Noodle, Lime, Onion, Carrot, Cilantro. Topped with Peanuts. Choice of Beef, Pork or Seafood.',
    price: 2100,
    category: 'Hot Meal',
    image: '6.jpg',
  },
  {
    id: 7,
    name: 'Yum - Seafood',
    description: 'Slice Thin Beef Mixed With Lettuce, Thread Noodle, Lime, Onion, Carrot, Cilantro. Topped with Peanuts. Choice of Beef, Pork or Seafood.',
    price: 1000,
    category: 'Lunch',
    image: '7.jpg',
  },
  {
    id: 8,
    name: 'Larb - Ground Beef',
    description: 'Ground Beef/Chicken/Pork mixed with Lemon Grass, Lemon Leaves, Onion, Cilantro, Green Onion, Basicl, and Ground Rice. Served with Cucumber and Cabbage/Lettuce.',
    price: 1300,
    category: 'Snack',
    image: '8.jpg',
  },
  {
    id: 9,
    name: 'Papaya Salad',
    description: 'Shredded Papaya, Lime, Pepper, Tomato, Dried Shrimp, Long Bean, Thai Eggplant, Fish Sauce, Peanuts. Served with Cabbage or Lettuce',
    price: 1000,
    category: 'Snack',
    image: '9.jpg',
  },
  {
    id: 10,
    name: 'Larb - Chicken',
    description: 'Ground Beef/Chicken/Pork mixed with Lemon Grass, Lemon Leaves, Onion, Cilantro, Green Onion, Basicl, and Ground Rice. Served with Cucumber and Cabbage/Lettuce.',
    price: 1200,
    category: 'Snack',
    image: '10.jpg',
  },
];


module.exports = meals;

