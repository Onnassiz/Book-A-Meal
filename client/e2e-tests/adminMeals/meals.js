/* eslint func-names: 0, no-unused-expressions: 0 */
const path = require('path');

const imagePath = path.join(__dirname, '../../assets/images/3.jpg');

const meals = {
  'Should sign user into the application': (client) => {
    client
      .url('http://localhost:3002/#/')
      .waitForElementVisible('header')
      .moveToElement('#login', 10, 10)
      .click('#login')
      .waitForElementVisible('h3')
      .assert.containsText('h3', 'Sign In')
      .setValue('#signInEmail', 'onnassiz@gmail.com')
      .setValue('#signInPassword', 'password')
      .click('input[type=submit]')
      .waitForElementVisible('header')
      .assert.visible('#caterer_dropdown')
      .assert.visible('#user_dropdown');
  },

  'Should navigate to the admin meals page': (client) => {
    client
      .moveToElement('#caterer_dropdown', 10, 10)
      .pause(1000)
      .waitForElementVisible('#admin_meals')
      .assert.visible('#admin_meals')
      .click('#admin_meals')
      .waitForElementVisible('#addMeal')
      .pause(2000)
      .assert.visible('#addMeal');
  },

  'Should show new meal modal form': (client) => {
    client
      .moveToElement('#addMeal', 10, 10)
      .pause(1000)
      .assert.containsText('#addMeal', 'Add Meal')
      .click('#addMeal')
      .pause(1000)
      .waitForElementVisible('.box')
      .assert.visible('.box')
      .pause(2000);
  },

  'Should fill meal form and create a new meal': (client) => {
    client
      .pause(1000)
      .setValue('#name', 'Fried Rice')
      .setValue('#price', 2000)
      .setValue('#category', 'Dinner')
      .setValue('#description', 'This is a good meal. It has all you need. Try it out.')
      .moveToElement('input[type=submit]', 10, 10)
      .pause(2000)
      .click('input[type=submit]')
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 5000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default');
  },

  'Should show a new meal modal form again': (client) => {
    client
      .moveToElement('#addMeal', 10, 10)
      .assert.containsText('#addMeal', 'Add Meal')
      .click('#addMeal')
      .waitForElementVisible('.box')
      .assert.visible('.box')
      .pause(1000);
  },

  'Should fill another meal form and create a new meal': (client) => {
    client
      .pause(1000)
      .setValue('#name', 'Chicken')
      .setValue('#price', 2000)
      .setValue('#category', 'Dinner')
      .setValue('#description', 'This is a good meal. It has all you need. Try it out.')
      .moveToElement('input[type=submit]', 10, 10)
      .pause(2000)
      .click('input[type=submit]')
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 5000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default');
  },

  'Should show a new meal modal form for the third time': (client) => {
    client
      .moveToElement('#addMeal', 10, 10)
      .assert.containsText('#addMeal', 'Add Meal')
      .click('#addMeal')
      .waitForElementVisible('.box')
      .assert.visible('.box')
      .pause(1000);
  },

  'Should fill the third meal form and create a new meal': (client) => {
    client
      .pause(1000)
      .setValue('#name', 'Beans')
      .setValue('#price', 1000)
      .setValue('#category', 'Dinner')
      .setValue('#description', 'This is a good meal for me and you.')
      .moveToElement('input[type=submit]', 10, 10)
      .pause(2000)
      .click('input[type=submit]')
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 5000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default');
  },

  'Should show an update meal modal': (client) => {
    client
      .waitForElementVisible('#updateModal:first-of-type')
      .assert.visible('#updateModal:first-of-type')
      .click('#updateModal:first-of-type')
      .assert.visible('.box')
      .pause(1000);
  },

  'Should update the third meal form and update a new meal': (client) => {
    client
      .pause(1000)
      .clearValue('#name')
      .setValue('#name', 'Rice')
      .clearValue('#price')
      .setValue('#price', 1200)
      .moveToElement('input[type=submit]', 10, 10)
      .pause(1000)
      .click('input[type=submit]')
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 5000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default');
  },

  'Should show delete modal and delete meal on button click': (client) => {
    client
      .assert.visible('#deleteModal:first-of-type')
      .click('#deleteModal:first-of-type')
      .waitForElementVisible('button.swal-button.swal-button--confirm.swal-button--danger', 10000)
      .pause(1000)
      .click('button.swal-button.swal-button--confirm.swal-button--danger')
      .waitForElementVisible('button.swal-button.swal-button--confirm', 15000)
      .pause(1000)
      .click('button.swal-button.swal-button--confirm')
      .waitForElementNotVisible('button.swal-button.swal-button--confirm', 20000)
      .assert.hidden('button.swal-button.swal-button--confirm')
      .pause(1000);
  },

  'Should show an image upload modal modal': (client) => {
    client
      .assert.visible('#addPhotoModal:first-of-type')
      .click('#addPhotoModal:first-of-type')
      .waitForElementVisible('#dropzoneHeader')
      .assert.containsText('#dropzoneHeader', 'Drop Image Here')
      .pause(1000)
      .moveToElement('#upload_image', 10, 10)
      .pause(1000)
      .setValue('input[type=file]', imagePath)
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 15000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default')
      .pause(1000);
    client.end();
  },
};

export default meals;
