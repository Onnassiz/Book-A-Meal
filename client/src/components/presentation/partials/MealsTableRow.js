import React from 'react';
import PropTypes from 'prop-types';
import empty from 'is-empty';
import { getDate, numberWithCommas } from '../../../utilities/functions';

const renderActionButtons = props => (
  <span>
    <span>
      <a title="edit" id="updateModal" onClick={() => props.toggleUpdateModal(props.item)}><i className="material-icons">mode_edit</i></a>
    </span>
    <span>
      <a title="add photo" id="addPhotoModal" onClick={() => props.toggleAddPhoto(props.item)}><i className="material-icons">add_a_photo</i></a>
    </span>
    <span>
      <a title="delete" id="deleteModal" onClick={() => props.toggleShowDeleteModal(props.item)}><i className="material-icons">delete_sweep</i></a>
    </span>
  </span>
);

const renderDetailTooltips = (props) => {
  let image = props.item.imageUrl;
  image = image === null ? null : image.replace('upload/', 'upload/c_scale,w_70/');
  return (
    <span>
      <a className="tooltips"><i className="material-icons">info_outline</i>
        <div className="arrowBox">
          <div>
            <p>
              {empty(props.item.imageUrl) ? <i style={{ fontSize: 80 }} className="material-icons">photo</i> : <img src={image} alt="banner" style={{ maxHeight: '600px', borderRadius: 4 }} />}
            </p>
            {props.item.description}
          </div>
        </div>
      </a>
    </span>
  );
};

const MealsTableRow = props => (
  <tr>
    <td style={{ width: 10 }}>{props.i}</td>
    <td>{props.item.name}</td>
    <td>&#8358;{numberWithCommas(props.item.price)}</td>
    <td>{props.item.category}</td>
    <td>{getDate(props.item.createdAt)}</td>
    <td>
      {renderDetailTooltips(props)}
    </td>
    <td>
      {renderActionButtons(props)}
    </td>
  </tr>
);

MealsTableRow.propTypes = {
  item: PropTypes.object.isRequired,
  i: PropTypes.number.isRequired,
};

renderDetailTooltips.propTypes = {
  item: PropTypes.object.isRequired,
};

renderActionButtons.propTypes = {
  toggleUpdateModal: PropTypes.func.isRequired,
  toggleShowDeleteModal: PropTypes.func.isRequired,
  toggleAddPhoto: PropTypes.func.isRequired,
  item: PropTypes.object.isRequired,
};


export default MealsTableRow;

