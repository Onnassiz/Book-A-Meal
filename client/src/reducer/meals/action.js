import axios from 'axios';
import { store } from '../../store';
import { setLoading, unsetLoading } from '../formState/action';

export const SET_MEAL = 'SET_MEAL';
export const SET_MEAL_SERVER_ERRORS = 'SET_MEAL_SERVER_ERRORS';
export const SET_MEAL_ALERT = 'SET_MEAL_ALERT';

function setAlert(alert) {
	return dispatch => dispatch({
		type: SET_MEAL_ALERT,
		alert,
	});
}

export function setMeal(meal) {
	let { meals } = store.getState().meals;
	meals = [meal].concat(meals);
	return dispatch => dispatch({
		type: SET_MEAL,
		meals,
	});
}

export function setMealArray(meal) {
	let { meals } = store.getState().meals;
	meals = meals.concat(meal);
	return dispatch => dispatch({
		type: SET_MEAL,
		meals,
	});
}

export function updateMealState(meal) {
	const { meals } = store.getState().meals;
	const index = meals.findIndex(x => x.id === meal.id);
	meals[index] = meal;
	return dispatch => dispatch({
		type: SET_MEAL,
		meals,
	});
}

export function updateMealAfterDelete(id) {
	let { meals } = store.getState().meals;
	meals = meals.filter(x => x.id !== id);
	return dispatch => dispatch({
		type: SET_MEAL,
		meals,
	});
}

function setMealErrors(errors) {
	return dispatch => dispatch({
		type: SET_MEAL_SERVER_ERRORS,
		errors,
	});
}

export function postMeal(meal) {
	return (dispatch) => {
		dispatch(setLoading());
		return axios.post('api/v1/meals', meal).then((response) => {
			dispatch(unsetLoading());
			dispatch(setMeal(response.data));
			dispatch(setAlert('Meal successfully created'));
			return response;
		}).catch((error) => {
			dispatch(unsetLoading());
			dispatch(setMealErrors(error.response));
			return error;
		});
	};
}

export function updateMeal(meal) {
	return (dispatch) => {
		dispatch(setLoading());
		return axios.put(`api/v1/meals/${meal.id}`, meal).then((response) => {
			dispatch(unsetLoading());
			dispatch(updateMealState(response.data));
			dispatch(setAlert('Meal successfully updated'));
			return response;
		}).catch((error) => {
			dispatch(unsetLoading());
			dispatch(setMealErrors(error.response.data));
			return error;
		});
	};
}

export function deleteMealById(id) {
	return (dispatch) => {
		dispatch(setLoading());
		return axios.delete(`api/v1/meals/${id}`).then((response) => {
			dispatch(unsetLoading());
			dispatch(updateMealAfterDelete(id));
			dispatch(setAlert('Meal successfully deleted'));
			return response;
		}).catch((error) => {
			dispatch(unsetLoading());
			dispatch(setMealErrors(error.response.data));
			return error;
		});
	};
}

export function putImage(id, field) {
	return (dispatch) => {
		dispatch(setLoading());
		return axios.put(`api/v1/meals/image/${id}`, field).then((response) => {
			dispatch(unsetLoading());
			dispatch(updateMealState(response.data));
			dispatch(setAlert('Image successfully uploaded'));
			return response;
		}).catch((error) => {
			dispatch(unsetLoading());
			dispatch(setMealErrors(error.response.data));
			return error;
		});
	};
}
export function getMeals() {
	return (dispatch) => {
		dispatch(setLoading());
		return axios.get('api/v1/meals').then((response) => {
			dispatch(unsetLoading());
			dispatch(setMealArray(response.data));
			return response;
		}).catch((error) => {
			dispatch(unsetLoading());
			return error;
		});
	};
}
