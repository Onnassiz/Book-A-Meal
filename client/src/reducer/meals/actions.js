import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actions';

export const SET_MEAL = 'SET_MEAL';
export const SET_MEAL_SERVER_ERRORS = 'SET_MEAL_SERVER_ERRORS';
export const SET_MEAL_ALERT = 'SET_MEAL_ALERT';

function setAlert(alert) {
  return dispatch => dispatch({
    type: SET_MEAL_ALERT,
    alert,
  });
}

// function setMeal(storeMeals, meal) {
//   const meals = [meal].concat(storeMeals);
//   return dispatch => dispatch({
//     type: SET_MEAL,
//     meals,
//   });
// }

// function setMealArray(storeMeals, meal) {
//   const meals = storeMeals.concat(meal);
//   return dispatch => dispatch({
//     type: SET_MEAL,
//     meals,
//   });
// }

export function updateMealState(storeMeals, meal) {
  const index = storeMeals.findIndex(x => x.id === meal.id);
  const meals = storeMeals;
  meals[index] = meal;
  return dispatch => dispatch({
    type: SET_MEAL,
    meals,
  });
}

export function updateMealAfterDelete(storeMeals, id) {
  const meals = storeMeals.filter(x => x.id !== id);
  return dispatch => dispatch({
    type: SET_MEAL,
    meals,
  });
}

function setMealErrors(errors) {
  return dispatch => dispatch({
    type: SET_MEAL_SERVER_ERRORS,
    errors,
  });
}

export function postMeal(meal) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.post('api/v1/meals', meal).then((response) => {
      dispatch(unsetLoading());
      // dispatch(setMeal(getState().meals.meals, response.data));
      dispatch(setAlert('Meal successfully created'));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMealErrors(error.response.data));
      return error;
    });
  };
}

export function updateMeal(meal) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.put(`api/v1/meals/${meal.id}`, meal).then((response) => {
      dispatch(unsetLoading());
      // dispatch(updateMealState(getState().meals.meals, response.data));
      dispatch(setAlert('Meal successfully updated'));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMealErrors(error.response.data));
      return error;
    });
  };
}

export function deleteMealById(id) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.delete(`api/v1/meals/${id}`).then((response) => {
      dispatch(unsetLoading());
      // dispatch(updateMealAfterDelete(getState().meals.meals, id));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMealErrors(error.response.data));
      return error;
    });
  };
}

export function getMeals(limit = 10, offset = 0) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.get(`api/v1/meals/?limit=${limit}&offset=${offset}`).then((response) => {
      dispatch(unsetLoading());
      // dispatch(setMealArray(getState().meals.meals, response.data));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      return error;
    });
  };
}
