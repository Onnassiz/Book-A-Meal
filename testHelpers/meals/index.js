
import { meal } from '../../api/models';
import { getCatererId } from '../main';
import getMealsMock from './mock';

const deleteMeals = (done, callDone = true) => meal.destroy({
  where: {},
}).then((deleted) => {
  if (callDone) {
    done();
  }
  return deleted;
});

const insertMealMock = done => getCatererId(done, false).then((id) => {
  const mocks = getMealsMock(id);

  return meal.bulkCreate(mocks).then(data => data);
});

const getMealId = () => meal.findOne({ where: { price: 80 } }).then(data => data.id);

const insertOneMeal = done => getCatererId(done, false).then((id) => {
  const newMeal = meal.build({
    name: 'The Good Meal',
    price: 2000,
    userId: id,
    description: null,
    category: 'Hot meal',
    imageUrl: 'http://bens.com',
  });

  return newMeal.save().then(data => data);
});

export {
  deleteMeals,
  insertOneMeal,
  getMealId,
  insertMealMock,
};

