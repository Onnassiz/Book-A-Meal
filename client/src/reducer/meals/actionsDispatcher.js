import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actionsDispatcher';
import { setMealErrorsAction } from './actions';

const setMealErrors = errors => dispatch => dispatch(setMealErrorsAction(errors));

export const postMeal = meal => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/meals', meal).then((response) => {
    dispatch(unsetLoading());
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setMealErrors(error.response.data));
    return error;
  });
};

export const updateMeal = meal => (dispatch) => {
  dispatch(setLoading());
  return axios.put(`api/v1/meals/${meal.id}`, meal).then((response) => {
    dispatch(unsetLoading());
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setMealErrors(error.response.data));
    return error;
  });
};

export const deleteMealById = id => (dispatch) => {
  dispatch(setLoading());
  return axios.delete(`api/v1/meals/${id}`).then((response) => {
    dispatch(unsetLoading());
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setMealErrors(error.response.data));
    return error;
  });
};

export const getMeals = (limit = 10, offset = 0) => (dispatch) => {
  dispatch(setLoading());
  return axios.get(`api/v1/meals/?limit=${limit}&offset=${offset}`).then((response) => {
    dispatch(unsetLoading());
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setMealErrors(error.response.data));
    return error;
  });
};
