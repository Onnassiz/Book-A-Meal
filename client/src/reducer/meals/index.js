import { SET_MEAL, SET_MEAL_SERVER_ERRORS } from './actions';

const initialState = {
  meals: [],
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_MEAL:
      return { ...state, meals: action.meals };
    case SET_MEAL_SERVER_ERRORS:
      return { ...state, errors: action.errors };
    default:
      return state;
  }
};
