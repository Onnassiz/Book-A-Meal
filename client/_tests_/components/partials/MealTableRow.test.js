import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MealsTableRow from '../../../src/components/presentation/partials/MealsTableRow';
import { singleMeal } from '../../objectProps/meals.props';

const props = {
  toggleAddPhoto: jest.fn(),
  toggleShowDeleteModal: jest.fn(),
  toggleUpdateModal: jest.fn(),
  item: singleMeal,
  key: singleMeal.id,
  i: 1,
};

const wrapper = shallow(<MealsTableRow {...props} />);

describe('<MealsTableRow />', () => {
  it('Should render table row component successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should it should call function when update modal is clicked', () => {
    wrapper.find('#updateModal').simulate('click');
    wrapper.find('#addPhotoModal').simulate('click');
    wrapper.find('#deleteModal').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
