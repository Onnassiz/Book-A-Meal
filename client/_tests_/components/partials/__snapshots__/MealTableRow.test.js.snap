// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealsTableRow /> Should it should call function when update modal is clicked 1`] = `
<tr>
  <td
    style={
      Object {
        "width": 10,
      }
    }
  >
    1
  </td>
  <td>
    Malvaceae
  </td>
  <td>
    ₦
    400
  </td>
  <td>
    dolor sit
  </td>
  <td>
    undefined, undefined NaN, NaN NaN:NaN
  </td>
  <td>
    <span>
      <a
        className="tooltips"
      >
        <i
          className="material-icons"
        >
          info_outline
        </i>
        <div
          className="arrowBox"
        >
          <div>
            <p>
              <img
                alt="banner"
                src="http://thisImage.jpeg"
                style={
                  Object {
                    "borderRadius": 4,
                    "maxHeight": "600px",
                  }
                }
              />
            </p>
            Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.
          </div>
        </div>
      </a>
    </span>
  </td>
  <td>
    <span>
      <span>
        <a
          id="updateModal"
          onClick={[Function]}
          title="edit"
        >
          <i
            className="material-icons"
          >
            mode_edit
          </i>
        </a>
      </span>
      <span>
        <a
          id="addPhotoModal"
          onClick={[Function]}
          title="add photo"
        >
          <i
            className="material-icons"
          >
            add_a_photo
          </i>
        </a>
      </span>
      <span>
        <a
          id="deleteModal"
          onClick={[Function]}
          title="delete"
        >
          <i
            className="material-icons"
          >
            delete_sweep
          </i>
        </a>
      </span>
    </span>
  </td>
</tr>
`;

exports[`<MealsTableRow /> Should render table row component successfully 1`] = `
<tr>
  <td
    style={
      Object {
        "width": 10,
      }
    }
  >
    1
  </td>
  <td>
    Malvaceae
  </td>
  <td>
    ₦
    400
  </td>
  <td>
    dolor sit
  </td>
  <td>
    undefined, undefined NaN, NaN NaN:NaN
  </td>
  <td>
    <span>
      <a
        className="tooltips"
      >
        <i
          className="material-icons"
        >
          info_outline
        </i>
        <div
          className="arrowBox"
        >
          <div>
            <p>
              <img
                alt="banner"
                src="http://thisImage.jpeg"
                style={
                  Object {
                    "borderRadius": 4,
                    "maxHeight": "600px",
                  }
                }
              />
            </p>
            Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.
          </div>
        </div>
      </a>
    </span>
  </td>
  <td>
    <span>
      <span>
        <a
          id="updateModal"
          onClick={[Function]}
          title="edit"
        >
          <i
            className="material-icons"
          >
            mode_edit
          </i>
        </a>
      </span>
      <span>
        <a
          id="addPhotoModal"
          onClick={[Function]}
          title="add photo"
        >
          <i
            className="material-icons"
          >
            add_a_photo
          </i>
        </a>
      </span>
      <span>
        <a
          id="deleteModal"
          onClick={[Function]}
          title="delete"
        >
          <i
            className="material-icons"
          >
            delete_sweep
          </i>
        </a>
      </span>
    </span>
  </td>
</tr>
`;
